src/Aspire.Hosting.Azure/IAddRoleAssignmentsContext.cs

```diff
@@ -38,4 +38,9 @@ public interface IAddRoleAssignmentsContext
     /// Not all role assignments require/use a principal name.
     /// </remarks>
     public BicepValue<string> PrincipalName { get; }
+
+    /// <summary>
+    /// Gets the execution context for distributed application.
+    /// </summary>
+    public DistributedApplicationExecutionContext ExecutionContext { get; }
```

Comment:
It might be more forward looking if we give the whole `IServiceProvider` here, so someone can get any service they want (if necessary).

Reply:
> It's exposed in DistributedApplicationExecutionContext,

Good call.